### PR TITLE
perl530: init at 5.30.0

### PR DESCRIPTION
**Pull request by volth**
###### Motivation for this change

new version

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by volth** [comment]
@GrahamcOfBorg build perl530
@GrahamcOfBorg build perl530.pkgs.LWP
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.perl530
@GrahamcOfBorg build pkgsCross.armv7l-hf-multiplatform.perl530.pkgs.LWP

**Comment by volth** [comment]
```perldevel``` is to be changed to ```perl-5.31.0``` but it is not supported by ```perl-cross``` yet.

**Comment by volth** [comment]
@rycee @aanderse @srhb

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -13891,9 +13891,10 @@ in
   ### DEVELOPMENT / PERL MODULES
 
   perlInterpreters = callPackages ../development/interpreters/perl {};
-  inherit (perlInterpreters) perl528 perldevel;
+  inherit (perlInterpreters) perl528 perl530 perldevel;
 
   perl528Packages = recurseIntoAttrs perl528.pkgs;
+  perl530Packages = recurseIntoAttrs perl530.pkgs;
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: Do we want to recurse into both sets? They will both be build by Hydra and show up with `nix-env`.

Comment (volth) [reply pkgs/top-level/all-packages.nix]: build by Hydra is desirable, it is going to be default ```pkgs.perl``` in about a month, so comparison of ```pkgs.perl528Packages``` and ```pkgs.perl530Packages``` would be useful for making decision about switching to ```perl530```.

About ```nix-env``` I have no opinion.

**Comment by aanderse** [comment]
@volth I hope to be able to test this in the coming week. Thanks!

**Pull request merged**